packages/core/src/components/forms/numericInput.tsx

```diff
@@ -107,6 +119,8 @@ export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInp
         buttonPosition: Position.RIGHT,
         majorStepSize: 10,
         minorStepSize: 0.1,
+        selectAllOnFocus: false,
+        selectAllOnIncrement: false,
```

Comment:
Actually, both `EditableText` and `TimePicker` expose a `selectAllOnFocus` prop, and both default the prop to `false`. I'd prefer to stay consistent with their APIs by defaulting `NumericInput`'s `selectAllOnFocus` to `false` too. Thoughts?